lib/svrquery/protocol/interfaces.go

```diff
@@ -23,6 +23,7 @@ type Mapper interface {
 // Client is an interface which is implemented by types which can act a query transport.
 type Client interface {
 	io.ReadWriteCloser
+	Queryer
```

Comment:
Out of interest, is this just usability? I see we populate this from the  protocol.Get(...) call so seems sensible but just checking

Reply:
Mainly since I now have 2 types of clients, the interface was needed so that `main.go` can use them as both `io.ReadWriteCloser` and `Queryer` (the UDP client implementation already implemented both). But I'll rework the client to make it backwards compatible so this might change again.